components/dashboard/src/AppNotifications.tsx

```diff
@@ -19,15 +19,22 @@ export function AppNotifications() {
             setNotifications(localState);
             return;
         }
-        (async () => {
-            const serverState = await getGitpodService().server.getNotifications();
-            setNotifications(serverState);
-            if (serverState.length > 0) {
-                setLocalStorageObject(KEY_APP_NOTIFICATIONS, serverState, /* expires in */ 60 /* seconds */);
-            }
-        })();
+        reloadNotifications().catch(console.error);
+
+        getGitpodService().registerClient({
+            onSubscriptionUpdate: () => reloadNotifications().catch(console.error),
```

Comment:
:cloud: Feels a bit off that we have the "notifications" abstraction but need to know about "subscription updates" and understand them to know when to reload notifications.

:bulb: Did you think about pushing notifications instead? Either the messages, or alternatively just the signal to reload (might be the cleaner solution, actually). That way, `subscriptionUpdate` could stay backend only.